### PR TITLE
[Python API] Rename configuration API + update tests/tools

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - get_metric, get_config -> get_property
 - set_config -> set_property
 - Update tests, pot, benchmark_app

### Tickets:
 - 76185

### CODE REVIEW

**Comment by akuporos** [comment]
@vladimir-dudnik please, estimate do python demos are affected by this renaming PR?

**Review by AlexeyLebedev1 — APPROVED** [review]

File: samples/python/hello_query_device/hello_query_device.py [hunk 1]
```diff
@@ -27,19 +27,19 @@ def main():
     for device in core.available_devices:
         log.info(f'{device} :')
         log.info('\tSUPPORTED_METRICS:')
-        for metric in core.get_metric(device, 'SUPPORTED_METRICS'):
+        for metric in core.get_property(device, 'SUPPORTED_METRICS'):
```
Comment (ilya-lavrenov) [inline samples/python/hello_query_device/hello_query_device.py:30]: 1. metric => property?
2. will you create variable for `SUPPORTED_METRICS` like it's done in C++?

File: src/bindings/python/src/pyopenvino/core/core.cpp [hunk 2]
```diff
@@ -27,6 +27,7 @@ void regclass_Core(py::module m) {
 
     cls.def(py::init<const std::string&>(), py::arg("xml_config_file") = "");
 
+    // todo: remove after Accuracy Checker migration to set/get_property API
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/pyopenvino/core/core.cpp:30]: is it possible to deprecate method in pybind?

File: src/bindings/python/src/pyopenvino/core/core.cpp [hunk 3]
```diff
@@ -35,6 +36,21 @@ void regclass_Core(py::module m) {
...
+        "set_property",
+        [](ov::Core& self, const std::map<std::string, std::string>& properties) {
+            self.set_property({properties.begin(), properties.end()});
+        },
+        py::arg("properties"));
+
+    cls.def(
+        "set_property",
+        [](ov::Core& self, const std::string& device_name, const std::map<std::string, std::string>& properties) {
+            self.set_property(device_name, {properties.begin(), properties.end()});
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/core.cpp:52]: Let's think about reusing `PyAny` wrapper class to replace `std::map<std::string, std::string>` with AnyMap-like construct of `std::map<std::string, ov::Any>`.

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp [hunk 4]
```diff
@@ -40,13 +40,29 @@ void regclass_CompiledModel(py::module m) {
...
         [](ov::CompiledModel& self, const std::string& name) -> py::object {
             return Common::from_ov_any(self.get_property(name)).as<py::object>();
         },
         py::arg("name"));
 
+    cls.def(
+        "get_property",
+        [](ov::CompiledModel& self, const std::string& name) -> py::object {
+            return Common::from_ov_any(self.get_property(name)).as<py::object>();
+        },
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/compiled_model.cpp]:
```suggestion
        "get_property",
        [](ov::CompiledModel& self, const std::string& name) {
            return Common::from_ov_any(self.get_property(name));
        },
```
`from_ov_any` is returning actual simple types of different classes (like int, float, vectors, map, tuples) which would be automatically converted to Python object by pybind's mechanism. If not please specify case/cases and extend common helper if possible.

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp [hunk 5]
```diff
@@ -40,13 +40,29 @@ void regclass_CompiledModel(py::module m) {
...
+        py::arg("properties"));
+
+    // todo: remove after Accuracy Checker migration to set/get_property API
     cls.def(
         "get_config",
         [](ov::CompiledModel& self, const std::string& name) -> py::object {
             return Common::from_ov_any(self.get_property(name)).as<py::object>();
         },
         py::arg("name"));
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/compiled_model.cpp]: Are we keeping it? I cannot see it in current header for CompiledModel class.

File: src/bindings/python/src/pyopenvino/core/core.cpp (hunk 2 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/core.cpp:30]: yep, done

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp (hunk 5 above)
Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp]: If yes, same comment as other `from_ov_any` and simple types.

File: src/bindings/python/src/pyopenvino/core/core.cpp [hunk 6]
```diff
@@ -139,6 +155,16 @@ void regclass_Core(py::module m) {
         py::arg("device_name"),
         py::arg("name"));
 
+    // todo: remove after Accuracy Checker migration to set/get_property API
+    cls.def(
+        "get_property",
+        [](ov::Core& self, const std::string& device_name, const std::string& name) -> py::object {
+            return Common::from_ov_any(self.get_property(device_name, name)).as<py::object>();
+        },
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/core.cpp:169]: Same comment as other `from_ov_any` and simple types. And possible solution:
```suggestion
        [](ov::Core& self, const std::string& device_name, py::object& name) {
            return Common::from_ov_any(self.get_property(device_name, PyAny(name)));
        },
```

**Review by jiwaszki — CHANGES_REQUESTED** [review]

**Comment by Wovchena** [comment]
> @vladimir-dudnik @Wovchena please, estimate do python demos are affected by this renaming PR?

Demos are going to be affected, but not merged yet:
https://github.com/openvinotoolkit/open_model_zoo/pull/3080/files#diff-f04b6126e0bc67db4160772abb17c33b9c4e023ce1842e0dbd7402e327eb7aecR71

+@akorobeinikov

Accuracy checker is affected, but you already know that.

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp (hunk 5 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp]: We keep it till OMZ will migrate to the new API in order not to break the CI.

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp]: I added deprecationWarning and 'todo comment' to remove it after OMZ migration

File: samples/python/hello_query_device/hello_query_device.py (hunk 1 above)
Comment (akuporos) [reply samples/python/hello_query_device/hello_query_device.py]: 1. good catch, done
2. discussed offline

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp (hunk 4 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp]: removed

File: src/bindings/python/src/pyopenvino/core/core.cpp (hunk 6 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/core.cpp]: `py::object` -- removed

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/core.cpp]: done

File: src/bindings/python/src/pyopenvino/core/core.cpp (hunk 3 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/core.cpp]: done

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp [hunk 7]
```diff
@@ -40,17 +42,39 @@ void regclass_CompiledModel(py::module m) {
...
+            for (const auto& property : properties) {
+                properties_to_cpp[property.first] = PyAny(property.second);
+            }
+            self.set_property({properties_to_cpp.begin(), properties_to_cpp.end()});
+        },
+        py::arg("properties"));
+
+    // todo: remove after Accuracy Checker migration to set/get_property API
     cls.def(
         "get_config",
```
Comment (ilyachur) [inline src/bindings/python/src/pyopenvino/core/compiled_model.cpp:58]: Do we have a plan to remove `get_confg` methods?

**Review by ilyachur — APPROVED** [review]

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp (hunk 7 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp:58]: Yes, as OMZ migrates to the new configuration API (77081 will be completed), I'll remove `set/get_config` and `get_metric` methods.

File: tools/pot/openvino/tools/pot/engines/ie_engine.py [hunk 8]
```diff
@@ -303,12 +303,12 @@ def completion_callback(request, user_data):
                 start_time = time()
 
         progress_log_fn = logger.info if print_progress else logger.debug
-        self._ie.set_config({'CPU_THROUGHPUT_STREAMS': 'CPU_THROUGHPUT_AUTO', 'CPU_BIND_THREAD': 'YES'}, self._device)
```
Comment (KodiaqQ) [inline tools/pot/openvino/tools/pot/engines/ie_engine.py:306]: I think we should use the try-except condition for this place. We will update this in the future separately. cc @AlexKoff88

Comment (akuporos) [reply tools/pot/openvino/tools/pot/engines/ie_engine.py:306]: done

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**